cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -671,4 +707,34 @@ protected static void countStageMetrics(StageMetrics stageMetrics,
       return sqlEngine.transform(sqlContext);
     }));
   }
+
+  /**
+   * Try to write the output directly to the SQLEngineOutput registered by this engine.
+   * @param datasetName dataset to write
+   * @param sqlEngineOutput output instance created by this engine
+   * @return {@link SQLEngineJob<Boolean>} representing if the write operation succeded.
+   */
+  public SQLEngineJob<Boolean> write(String datasetName, SQLEngineOutput sqlEngineOutput) {
+    return runJob(datasetName, SQLEngineJobType.WRITE, () -> {
+      getDatasetForStage(datasetName);
+      LOG.debug("Attempting write for dataset {} into {}", datasetName, sqlEngineOutput);
```

Comment:
SQLEngineOutput does not have a good "toString". Also putting whole list of parameters (including potentially two jsons) into log is an overkill and may have passwords

Reply:
hmm, good catch. Let me add a better `toString` method